server/src/main/java/org/apache/druid/metadata/segment/cache/HeapMemorySegmentMetadataCache.java

```diff
@@ -870,11 +895,93 @@ private void retrieveAllPendingSegments(
     }
   }
 
+  /**
+   * Retrieves all segment schemas from the metadata store irrespective of their
+   * used status or last updated time.
+   */
+  private void retrieveAllSegmentSchemas(
+      Map<String, DatasourceSegmentSummary> datasourceToSummary
+  )
+  {
+    if (!useSchemaCache) {
+      return;
+    }
+
+    // Emit metrics for the current contents of the cache
+    segmentSchemaCache.getStats().forEach(this::emitMetric);
+
+    final Stopwatch fetchDuration = Stopwatch.createStarted();
+    final String sql = StringUtils.format(
+        "SELECT fingerprint, payload FROM %s WHERE version = %s",
+        tablesConfig.getSegmentSchemasTable(), CentralizedDatasourceSchemaConfig.SCHEMA_VERSION
+    );
```

Comment:
I think this code is fetching the entire set of segment schemas on every call to `syncWithMetadataStore`. Is this going to be OK, performance-wise? It seems expensive.

Reply:
The current implementation in `SqlSegmentsMetadataManager` polls all the schemas in every sync too.
But since the schemas table already has a `used_status_last_updated` as well as a `created_time` column,
we can try to do delta syncs in a fashion similar to the segments table.

Thanks for the suggestion, I will update the PR accordingly.